sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -309,93 +300,85 @@ namespace Azure { namespace Core { namespace Http {
      * @brief Append an element of URL path.
      *
      * @param path URL path element to append.
-     * @param isPathEncoded `true` if \p path is URL-encoded, `false` otherwise.
      */
-    void AppendPath(const std::string& path, bool isPathEncoded = false)
+    void AppendPath(const std::string& encodedPath)
     {
       if (!m_encodedPath.empty() && m_encodedPath.back() != '/')
       {
         m_encodedPath += '/';
       }
-      m_encodedPath += isPathEncoded ? path : EncodePath(path);
+      m_encodedPath += encodedPath;
     }
 
     /**
-     * @brief Append an HTTP query parameter.
+     * @brief The value of a query parameter is expected to be non-URL-encoded and, by default, it
+     * will be encoded before adding to the URL. Use \p isValueEncoded = true when the
+     * value is already encoded.
      *
-     * @note Overrides previous query parameters.
-     * @remark A query key can't contain any characters that need to be URL-encoded (per RFC 7230).
+     * @remark This function overrides the value of existing query parameters.
      *
-     * @param key HTTP query parameter.
-     * @param value HTTP query parameter value.
-     * @param isValueEncoded `true` if \p value is URL-encoded, `false` otherwise.
+     * @param encodedKey Name of the query parameter, already encoded.
+     * @param encodedValue Value of the query parameter, already encoded.
      */
-    void AppendQuery(const std::string& key, const std::string& value, bool isValueEncoded = false)
+    void AppendQueryParameter(const std::string& encodedKey, const std::string& encodedValue)
     {
-      std::string encoded_value = isValueEncoded ? Decode(value) : value;
-      if (m_retryModeEnabled)
-      {
-        m_retryQueryParameters[key] = encoded_value;
-      }
-      else
-      {
-        m_queryParameters[key] = encoded_value;
-      }
+      m_encodedQueryParameters[encodedKey] = encodedValue;
     }
 
     /**
-     * @brief Append a HTTP query.
+     * @brief Finds the first '?' symbol and parses everything after it as query parameters.
+     * separated by '&'.
      *
-     * @note All the required HTTP query parts should be URL-encoded.
-     *
-     * @param encodedQueries HTTP query.
+     * @param encodedQueryParameters String containing one or more query parameters.
```

Comment:
Fill this in, to explain the format:
```suggestion
     * @param encodedQueryParameters String containing one or more query parameters, separated by ...
```